R/verb-arrange.R

```diff
@@ -45,7 +50,16 @@ op_desc.op_arrange <- function(x, ...) {
 #' @export
 sql_build.op_arrange <- function(op, con, ...) {
   order_vars <- translate_sql_(op$dots, con, context = list(clause = "ORDER"))
-  group_vars <- c.sql(ident(op_grps(op$x)), con = con)
-
-  select_query(sql_build(op$x, con), order_by = order_vars)
+  if (op$args$.by_group) {
+    order_by <- c.sql(ident(op_grps(op$x)), order_vars, con = con)
+    select_query(
```

Comment:
I think you should move the `select_query()` call outside of the `if` block to reduce the amount of duplication.